`thread_mutex_trylock`: error: control reaches end of non-void function [-Werror=return-type]

**Describe the bug**

1.2.18 build fails when treating on-void returns as errors.

```
[   94s] gcc -DHAVE_CONFIG_H -I. -I./config     -I/usr/include/net-snmp -I/usr/include/net-snmp/.. -I/usr/include/mysql -O2 -Wall -D_FORTIFY_SOURCE=2 -fstack-protector-strong -funwind-tables -fasynchronous-unwind-tables -fstack-clash-protection -Werror=return-type -flto=auto -c -o locks.o locks.c
[   94s] locks.c: In function 'thread_mutex_trylock':
[   94s] locks.c:238:1: error: control reaches end of non-void function [-Werror=return-type]
[   94s]   238 | }
```

**To Reproduce**

Build with `-Werror=return-type`.

**Expected behavior**

The `int thread_mutex_trylock` function should actually return something.
See https://github.com/Cacti/spine/pull/223/files#diff-fde72ea96652c9cc626c72761023f4a5aefe6f49c5652ad9e026ad9d4a941775R264

(and the others)

**Screenshots**

N/A

**Server (please complete the following information):**
 - OS: openSUSE Tumbleweed
 - Version 2021-08-ish

**Compiling (please complete the following information):**

 - compiler: gcc 11
 - autoconf: 2.69
 - glibc: 2.33
 - source: all

**Additional context**

#223

--- a/CHANGELOG
+++ b/CHANGELOG
@@ -1,5 +1,8 @@
 The Cacti Group | spine
 
+1.2.19
+-issue#227: Fix non-void return in thread_mutex_trylock
+
 1.2.18
 -issue#207: Missing time parameter on FROM_UNIXTIME function
 -issue#221: IPv6 devices are not properly parsed

--- a/locks.c
+++ b/locks.c
@@ -235,4 +235,5 @@ int thread_mutex_trylock(int mutex) {
 	SPINE_LOG_DEVDBG(( "LOCKS: [START] Mutex try lock for %s", get_name(mutex) ));
 	int ret_val = pthread_mutex_trylock(get_lock(mutex));
 	SPINE_LOG_DEVDBG(( "LOCKS: [ END ] Mutex try lock for %s, result = %d", get_name(mutex), ret_val ));
+	return ret_val;
 }

